File: Python/8. Difference Between Element Sum and Digit Sum of an Array/difference.py
nums = [23, 456, 71]
digit_sum = 0
number_sum = 0
for num in nums:
    number_sum += num
    num_str = str(num)
    for digit_str in num_str:
        digit = int(digit_str)
        digit_sum += digit
print(abs(digit_sum - number_sum))

File: Java/7. Difference Between Element Sum and Digit Sum of an Array/Difference.java
public class Difference {
    public static int getElementSum(int[] arr) {
        int sum = 0;

        for (int num : arr) {
            sum += num;
        }

        return sum;
    }

    public static int getDigitSum(int[] arr) {
        int sum = 0;

        for (int num : arr) {
            String numStr = String.valueOf(num);

            for (char digitChar : numStr.toCharArray()) {
                int digit = Character.getNumericValue(digitChar);
                sum += digit;
            }
        }

        return sum;
    }

    public static void main(String[] args) {
        int[] array = { 23, 456, 71 };
        int elementSum = getElementSum(array);
        int digitSum = getDigitSum(array);
        int difference = elementSum - digitSum;

        System.out.println("Difference: " + difference);
    }
}
